File: backend/app/schemas/tipo_cultura.py
"""
Schemas de validação de entrada/saída do catálogo de Tipos de Cultura.
"""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TipoCulturaCreate(BaseModel):
    nome: str = Field(..., min_length=2, max_length=100)
    descricao: str | None = Field(default=None, max_length=300)


class TipoCulturaUpdate(BaseModel):
    nome: str | None = Field(default=None, min_length=2, max_length=100)
    descricao: str | None = Field(default=None, max_length=300)


class TipoCulturaOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nome: str
    descricao: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime


class TipoCulturaListOut(BaseModel):
    total: int
    items: list[TipoCulturaOut]

File: frontend/src/types/tipoCultura.ts
export interface TipoCultura {
  id: string;
  nome: string;
  descricao: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface TipoCulturaListagem {
  total: number;
  items: TipoCultura[];
}

export interface TipoCulturaFormData {
  nome: string;
  descricao?: string | null;
}
